src/control/lib/spdk/src/nvme_control.c

```diff
@@ -65,121 +71,204 @@ get_dev_health_logs(struct spdk_nvme_ctrlr *ctrlr,
 }
 
 struct ret_t *
-nvme_discover(bool init_vmd)
+nvme_discover(void)
 {
-	return _discover(&spdk_nvme_probe, true, &get_dev_health_logs,
-			 init_vmd);
+	return _discover(&spdk_nvme_probe, true, &get_dev_health_logs);
 }
 
 static void
-write_complete(void *arg, const struct spdk_nvme_cpl *completion)
+read_complete(void *arg, const struct spdk_nvme_cpl *completion)
 {
-	struct format_sequence	*sequence = arg;
+	struct wipe_sequence *sequence = arg;
 
-	if (sequence->using_cmb_io) {
-		spdk_nvme_ctrlr_free_cmb_io_buffer(sequence->ctrlr,
-						   sequence->buf, 0x1000);
-	} else {
-		spdk_free(sequence->buf);
+	if (spdk_nvme_cpl_is_success(completion)) {
+		sequence->is_completed = 1;
+		return;
 	}
-	sequence->is_completed = 1;
+
+	spdk_nvme_qpair_print_completion(sequence->ns_entry->qpair,
+					 (struct spdk_nvme_cpl *)completion);
+	fprintf(stderr, "I/O error status: %s\n",
+		spdk_nvme_cpl_get_status_string(&completion->status));
+	fprintf(stderr, "Read I/O failed, aborting run\n");
+	sequence->is_completed = 2;
```

Comment:
I had assumed this was acting as a bool based on the name. If it's an enum, let's name the values.

Reply:
done